src/os/bluestore/BlueStore.cc

```diff
@@ -12146,7 +12152,15 @@ ObjectMap::ObjectMapIterator BlueStore::get_omap_iterator(
   }
   o->flush();
   dout(10) << __func__ << " has_omap = " << (int)o->onode.has_omap() <<dendl;
-  KeyValueDB::Iterator it = db->get_iterator(o->get_omap_prefix());
+  auto bounds = KeyValueDB::IteratorBounds();
+  if (o->onode.has_omap()) {
+    std::string lower_bound, upper_bound;
+    o->get_omap_key(string(), &lower_bound);
+    o->get_omap_tail(&upper_bound);
+    bounds.lower_bound = lower_bound;
+    bounds.upper_bound = upper_bound;
+  }
+  KeyValueDB::Iterator it = db->get_iterator(o->get_omap_prefix(), 0, bounds);
```

Comment:
there are a several places that we're copying strings unnecessarily. these omap keys may tend to be large, so each copy would mean an extra memory allocation

prefer to move strings if you're done using them:
```c++
    bounds.lower_bound = std::move(lower_bound);
    bounds.upper_bound = std::move(upper_bound);
```
this goes for `KeyValueDB::IteratorBounds` too, which gets passed to functions by value:
```c++
  KeyValueDB::Iterator it = db->get_iterator(o->get_omap_prefix(), 0, std::move(bounds));
```